src/services/itemLogin/index.ts

```diff
@@ -34,9 +39,34 @@ const plugin: FastifyPluginAsync = async (fastify) => {
     '/:id/login-schema-type',
     { schema: getLoginSchemaType, preHandler: optionalIsAuthenticated },
     async ({ user, params: { id: itemId } }) => {
-      const value =
-        (await itemLoginService.getSchemaType(user?.account, buildRepositories(), itemId)) ?? null;
-      return value;
+      return await db.transaction(async (manager) => {
+        const repositories = buildRepositories(manager);
+        const item = await itemService.get(
+          user?.account,
+          repositories,
+          itemId,
+          PermissionLevel.Read,
+          false,
+        );
+
+        const tags = await itemTagService.getByItemPath(repositories, item.path);
+        const isHidden = Boolean(tags.find((t) => t.type === ItemTagType.Hidden));
+        if (
+          isHidden &&
+          (user?.account == undefined ||
+            !(await itemMembershipService.getByAccountAndItem(
+              repositories,
+              user?.account?.id,
+              item.id,
```

Comment:
Extract this in a variable to improve clarity ^^

Reply:
If I extract we'll always make a request to the Database. It could also expose us to some issues if this request fail and throws an exception when we don't need to.

I'll add some comments !